src/NonCumulative/Soundness/Properties/LogRel.agda

```diff
@@ -86,4 +88,620 @@ Glu-wellfounded-≡-GluΛ {j} {k} = implicit-extensionality fext (fext (λ l<k 
\ No newline at end of file
+®El⇒ty A≈B t∼a = ®⇒ty A≈B (®El⇒® A≈B t∼a)
+
+-- ®El respects term equivalence.
+®El-resp-≈ : ∀ {i} (A≈B : A ≈ B ∈ 𝕌 i) →
+             Γ ⊢ t ∶ T ®[ i ] a ∈El A≈B →
+             Γ ⊢ t ≈ t′ ∶[ i ] T →
+             ----------------------------
+             Γ ⊢ t′ ∶ T ®[ i ] a ∈El A≈B
+®El-resp-≈ (ne C≈C j≡1+i j′≡1+i) (ne c≈c′ refl _ , glu) t≈t′ = ne c≈c′ refl refl , record
+  { t∶T  = proj₁ (proj₂ (proj₂ (presup-≈ t≈t′)))
+  ; ⊢T   = ⊢T
+  ; krip = λ ⊢σ → proj₁ (krip ⊢σ) , ≈-trans ([]-cong (≈-sym t≈t′) (s-≈-refl (⊢w⇒⊢s ⊢σ))) (proj₂ (krip ⊢σ))
+  }
+  where open GluNe glu
+®El-resp-≈ N′ (t® , T≈N) t≈t′ = ®Nat-resp-≈ t® (≈-conv t≈t′ T≈N) , T≈N
+®El-resp-≈ (U′ {j}) record { t∶T = t∶T ; T≈ = T≈ ; A∈𝕌 = A∈𝕌 ; rel = rel } t≈t′
+  rewrite 𝕌-wf-gen j (λ l<j → <-trans l<j (s≤s (≤-reflexive refl)))
+        | Glu-wf-gen {j} j U≤′ = record
+    { t∶T = proj₁ (proj₂ (proj₂ (presup-≈ t≈t′)))
+    ; T≈  = T≈
+    ; A∈𝕌 = A∈𝕌
+    ; rel = ®-resp-≈ A∈𝕌 rel (≈-conv t≈t′ T≈)
+    }
+®El-resp-≈ {Γ = Γ} {t = t} {t′ = t′} (Π′ {j = j} {k = k} jA RT) record { t∶T = t∶T ; a∈El = a∈El ; IT = IT ; OT = OT ; ⊢IT = ⊢IT ; ⊢OT = ⊢OT ; T≈ = T≈ ; krip = krip } t≈t′
+  rewrite 𝕌-wf-gen k (ΠO≤′ j k refl)
+        | 𝕌-wf-gen j (ΠI≤′ j k refl)
+        | Glu-wf-gen j (ΠI≤′ j k refl)
+        | Glu-wf-gen k (ΠO≤′ j k refl) = record
+  { t∶T  = proj₁ (proj₂ (proj₂ (presup-≈ t≈t′)))
+  ; a∈El = a∈El
+  ; IT   = IT
+  ; OT   = OT
+  ; ⊢IT  = ⊢IT
+  ; ⊢OT  = ⊢OT
+  ; T≈   = T≈
+  ; krip = λ ⊢σ → let open ΛRel (krip ⊢σ) in record
+      { IT-rel = IT-rel
+      ; ap-rel = λ t® b∈ →
+        let open ΛKripke (ap-rel t® b∈)
+        in record
+          { fa = fa
+          ; ↘fa = ↘fa
+          ; ®fa = ®El-resp-≈ (ΠRT.T≈T′ (RT b∈)) ®fa (t[σ]≈t′[σ]s (⊢w⇒⊢s ⊢σ) (®El⇒tm jA t®))
+          }
+      }
+  }
+    where t[σ]≈t′[σ]s : Δ ⊢s σ ∶ Γ →
+               Δ ⊢ s ∶[ j ] IT [ σ ] →
+               ----------------------
+               Δ ⊢ t [ σ ] $ s ≈ t′ [ σ ] $ s ∶[ k ] OT [ σ , s ∶ IT ↙ j ]
+          t[σ]≈t′[σ]s ⊢σ′ ⊢s = ≈-conv ($-cong Δ⊢IT[σ] Δ⊢OT[σ]
+                                              (≈-conv ([]-cong t≈t′ (s-≈-refl ⊢σ′)) (≈-trans ([]-cong-Se′ T≈ ⊢σ′) (Π-[] ⊢σ′ ⊢IT ⊢OT refl)))
+                                              (≈-refl ⊢s) refl)
+                                      (≈-trans ([∘]-Se ⊢OT IT,Δ⊢s (⊢I,t ⊢Δ Δ⊢IT[σ] ⊢s))
+                                               ([]-cong-Se″ ⊢OT
+                                                            (s-∘ (s-, (s-I ⊢Δ) Δ⊢IT[σ] (conv ⊢s (≈-sym ([I] Δ⊢IT[σ])))) IT,Δ⊢s)
+                                                            (qσ∘[I,t]≈σ,t ⊢Δ ⊢IT ⊢σ′ ⊢s)))
+            where ⊢Δ      = (proj₁ (presup-s ⊢σ′))
+                  Δ⊢IT[σ] = t[σ]-Se ⊢IT ⊢σ′
+                  IT,Δ⊢s  = ⊢q ⊢Δ ⊢σ′ ⊢IT
+                  Δ⊢OT[σ] = t[σ]-Se ⊢OT IT,Δ⊢s
+
+®El-resp-≈ {i = i} (L {j = j} {k = k} refl kA refl refl) record { t∶T = t∶T ; UT = UT ; ⊢UT = ⊢UT ; a∈El = a∈El ; T≈ = T≈ ; krip = krip } t≈t′
```

Comment:
`L'`. emacs will automatically change your constructors to abbreviated forms. it seems VS code doesn't do it.